### PR TITLE
Features/rhpm lg01 01

### PR DESCRIPTION
**Pull request by karthikkharvi**
Service file and method for user login

### CODE REVIEW

File: api/RS.API/RS.Service/Logic/UserManagerService.cs [hunk 1]
```diff
@@ -20,6 +20,50 @@ public UserManagerService(IUserRepository userRepository)
...
+            };
+            try
+            {
+                var userModel = new Users();
+                userModel.MapFromViewModel(user);
+                var duplicateUser = _userRepository.GetFirstOrDefault(x => x.UserName == user.UserName || x.Email == user.Email);
+                if (duplicateUser != null)
+                {
+                    result.Status = Status.Fail;
+                    result.Message = SkillStatusNotification.DuplicateSkill;
```
Comment (salilsaini868) [inline api/RS.API/RS.Service/Logic/UserManagerService.cs]: Change message to user duplicate

File: web/src/app/Login/login.Component.ts [hunk 2]
```diff
@@ -1,14 +1,44 @@
...
 @Component({
   selector: 'login',
   templateUrl: 'login.component.html'
 })
 
 export class LoginComponent implements OnInit {
-  constructor() {
+
+  userEmail: string;
+  userPassword: string;
```
Comment (salilsaini868) [inline web/src/app/Login/login.Component.ts:14]: remove email, password variable, use user login model created from swagger

File: web/src/app/Login/login.Component.ts [hunk 3]
```diff
@@ -1,14 +1,44 @@
...
+  constructor(private loginService: LoginService) {
   }
 
   ngOnInit() {
   }
+
+  onSubmit(loginForm) {
+    if (loginForm.valid) {
+      let loginModel;
+      loginModel = { userEmail: this.userEmail, userPassword: this.userPassword } as UserLoginModel;
```
Comment (salilsaini868) [inline web/src/app/Login/login.Component.ts:28]: use direct user login model rather than casting

File: web/src/app/Login/login.Component.ts [hunk 4]
```diff
@@ -1,14 +1,44 @@
...
   templateUrl: 'login.component.html'
 })
 
 export class LoginComponent implements OnInit {
-  constructor() {
+
+  userEmail: string;
+  userPassword: string;
+  loginModel: UserLoginModel;
+  invalidCredentials : boolean;
```
Comment (salilsaini868) [inline web/src/app/Login/login.Component.ts:16]: why this invalidCredential variable is used, remove unnecessary variables.

File: web/src/app/Login/login.Component.ts [hunk 5]
```diff
@@ -1,14 +1,44 @@
...
+      this.loginService.userLogin(loginModel).subscribe(
+        (data) => {
+          if (!isNullOrUndefined(data)) {
+                localStorage.setItem("auth_token", data);
+          } else {
+            this.invalidCredentials = true;
+          }
+        },
+        (err) => {
+          this.errorMessage = err.message;
```
Comment (salilsaini868) [inline web/src/app/Login/login.Component.ts:38]: use toster for messages and remove vriables

File: web/src/app/Login/shared/login.service.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import 'rxjs/Rx';
+import { ApiClientService } from '../.././services/swagger-generated/apiClientService';
+
+@Injectable()
+export class LoginService {
+
+    constructor(private http: Http, private apiClient: ApiClientService) { }
+
+    userLogin(userLoginModel) : Observable<any> {
+        debugger;
```
Comment (salilsaini868) [inline web/src/app/Login/shared/login.service.ts:13]: remove debugger

File: web/src/app/app.module.ts [hunk 7]
```diff
@@ -26,7 +29,7 @@ import { SharedModule } from './shared/shared.module';
     SharedModule.forRoot()
     ],
   exports: [],
-  providers: [],
+  providers: [ApiClientService],
```
Comment (salilsaini868) [inline web/src/app/app.module.ts:32]: apiclientservice should be in corresponding module i.e. in service.module.ts

File: api/RS.API/RS.Service/Logic/UserManagerService.cs (hunk 1 above)
Comment (karthikkharvi) [reply api/RS.API/RS.Service/Logic/UserManagerService.cs]: Done

**Review by salilsaini868 — CHANGES_REQUESTED** [review]

File: web/src/app/Login/login.Component.ts (hunk 4 above)
Comment (karthikkharvi) [reply web/src/app/Login/login.Component.ts:16]: Done

File: web/src/app/Login/login.Component.ts (hunk 5 above)
Comment (karthikkharvi) [reply web/src/app/Login/login.Component.ts:38]: removed uneccessary variables

File: web/src/app/Login/shared/login.service.ts (hunk 6 above)
Comment (karthikkharvi) [reply web/src/app/Login/shared/login.service.ts:13]: done

File: web/src/app/app.module.ts (hunk 7 above)
Comment (karthikkharvi) [reply web/src/app/app.module.ts:32]: Done

File: web/src/app/Login/login.Component.ts (hunk 3 above)
Comment (karthikkharvi) [reply web/src/app/Login/login.Component.ts:28]: Done

**Review by salilsaini868 — APPROVED** [review]

**Pull request merged**